pkg/k8s/K8sCommonService.go

```diff
@@ -0,0 +1,464 @@
+	GetCoreClientByClusterId(clusterId int) (*kubernetes.Clientset, *v1.CoreV1Client, error)
+	GetK8sServerVersion(clusterId int) (*version.Info, error)
+}
+type K8sCommonServiceImpl struct {
+	logger                      *zap.SugaredLogger
+	K8sUtil                     *k8s.K8sUtil
+	helmAppService              client.HelmAppService
+	K8sResourceHistoryService   kubernetesResourceAuditLogs.K8sResourceHistoryService
+	clusterService              cluster.ClusterService
+	K8sApplicationServiceConfig *K8sApplicationServiceConfig
+}
+type K8sApplicationServiceConfig struct {
+	BatchSize        int `env:"BATCH_SIZE" envDefault:"5"`
+	TimeOutInSeconds int `env:"TIMEOUT_IN_SECONDS" envDefault:"5"`
+}
+
+func NewK8sCommonServiceImpl(Logger *zap.SugaredLogger, k8sUtils *k8s.K8sUtil, helmAppService client.HelmAppService, K8sResourceHistoryService kubernetesResourceAuditLogs.K8sResourceHistoryService, clusterService cluster.ClusterService) *K8sCommonServiceImpl {
+	cfg := &K8sApplicationServiceConfig{}
+	err := env.Parse(cfg)
+	if err != nil {
+		Logger.Infow("error occurred while parsing K8sApplicationServiceConfig,so setting batchSize and timeOutInSeconds to default value", "err", err)
+	}
+	return &K8sCommonServiceImpl{
+		logger:                      Logger,
+		K8sUtil:                     k8sUtils,
+		helmAppService:              helmAppService,
+		K8sResourceHistoryService:   K8sResourceHistoryService,
+		clusterService:              clusterService,
+		K8sApplicationServiceConfig: cfg,
+	}
+}
+
+func (impl *K8sCommonServiceImpl) GetResource(ctx context.Context, request *ResourceRequestBean) (*k8s.ManifestResponse, error) {
+	clusterId := request.ClusterId
+	//getting rest config by clusterId
+	restConfig, err, _ := impl.GetRestConfigByClusterId(ctx, clusterId)
+	if err != nil {
+		impl.logger.Errorw("error in getting rest config by cluster Id", "err", err, "clusterId", clusterId)
+		return nil, err
+	}
+	resourceIdentifier := request.K8sRequest.ResourceIdentifier
+	resp, err := impl.K8sUtil.GetResource(ctx, resourceIdentifier.Namespace, resourceIdentifier.Name, resourceIdentifier.GroupVersionKind, restConfig)
+	if err != nil {
+		impl.logger.Errorw("error in getting resource", "err", err, "resource", resourceIdentifier.Name)
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (impl *K8sCommonServiceImpl) CreateResource(ctx context.Context, request *ResourceRequestBean) (*k8s.ManifestResponse, error) {
+	resourceIdentifier := &openapi.ResourceIdentifier{
+		Name:      &request.K8sRequest.ResourceIdentifier.Name,
+		Namespace: &request.K8sRequest.ResourceIdentifier.Namespace,
+		Group:     &request.K8sRequest.ResourceIdentifier.GroupVersionKind.Group,
+		Version:   &request.K8sRequest.ResourceIdentifier.GroupVersionKind.Version,
+		Kind:      &request.K8sRequest.ResourceIdentifier.GroupVersionKind.Kind,
+	}
+	manifestRes, err := impl.helmAppService.GetDesiredManifest(ctx, request.AppIdentifier, resourceIdentifier)
```

Comment:
why are we using helmservice in k8scommonservice?

Reply:
we are checking desired manifest before creating resource which is written in kubelink. Need to think about ways where we can use common methods like this without breaking architecture standards

Reply:
It was used in case of external helm app when resource health goes missing, we were fetching desired manifest from kubelink to recreate resource, have refactored it to application service with appropriate name.